server/routes/api/users.js

```diff
@@ -87,4 +88,37 @@ router.get("/contests", (req, res) => {
     });
 });
 
+// @route   GET api/user/submissions/
+// @desc    Get all contests that a user submitted to
+// @access  Private
+router.get("/submissions", auth, (req, res) => {
+  const { user_id } = req.query;
+  let result = {};
+
+  Submission.find({ user_id })
+    .then((submissions) => {
+      let contestIds = [];
+      submissions.forEach(submission => contestIds.push(submission.contest_id));
+      result.submissions = submissions;
+      Contest.find({
```

Comment:
i wonder if you can use an aggregator to simplify this query?

https://docs.mongodb.com/manual/reference/operator/aggregation/lookup/